include/vcpkg/base/messages.h

```diff
@@ -148,16 +148,66 @@ namespace vcpkg::msg
 
         std::string format_examples_for_args(StringView extra_comment, const FormatArgAbi* args, std::size_t arg_count);
 
-        inline std::string get_examples_for_args(StringView extra_comment, const MessageCheckFormatArgs<>&)
+        template<::size_t M, ::size_t N>
+        inline std::string get_examples_for_args(const StringArray<M>& comment, const StringArray<N>& example)
         {
-            return extra_comment.to_string();
+            if (comment.length() == 0 && example.length() == 0)
+            {
+                return std::string{""};
+            }
+            if (comment.length() == 0)
+            {
+                return std::string(example.begin(), example.end() - 1);
+            }
+
+            if (example.length() == 0)
+            {
+                return std::string(comment.begin(), comment.end() - 1);
+            }
+
+            if (example[0] == ' ')
+            {
+                const auto out = comment + example;
+                return std::string(out.begin(), out.end() - 1);
+            }
+
+            const auto out = comment + StringArray(" ") + example;
+            return std::string(out.begin(), out.end() - 1);
+        }
+
+        template<class Arg0>
+        constexpr auto example_piece(const Arg0& arg)
+        {
+            if constexpr (Arg0::real_example().length() == 0)
+                return StringArray{""};
+            else
+                return StringArray{" "} + arg.real_example();
         }
 
         template<class Arg0, class... Args>
-        std::string get_examples_for_args(StringView extra_comment, const MessageCheckFormatArgs<Arg0, Args...>&)
+        constexpr auto example_piece(const Arg0& arg, Args... args)
```

Comment:
This is basically the same as `Strings::concat`: It concatenates the example strings and inserts spaces where necessary.